scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/model/DefaultModelJobValidatorFactory.java

```diff
@@ -0,0 +1,391 @@
+                    return;
+                case LONG_MODEL:
+                    checkValueOf(task, variable, model, new Long(0));
+                    return;
+                case DOUBLE_MODEL:
+                    checkValueOf(task, variable, model, new Double(0));
+                    return;
+                case BOOLEAN_MODEL:
+                    checkBoolean(task, variable, model);
+                    return;
+                case FLOAT_MODEL:
+                    checkValueOf(task, variable, model, new Float(0));
+                    return;
+                case SHORT_MODEL:
+                    checkValueOf(task, variable, model, new Short((short) 0));
+                    return;
+                case URL_MODEL:
+                    checkURL(task, variable, model);
+                    return;
+                case URI_MODEL:
+                    checkURI(task, variable, model);
+                    return;
+                default:
+                    // not one of the above models
+            }
+
+            if (upperCaseModel.startsWith(INTEGER_MODEL)) {
+                checkPrimitiveValueInRange(task, variable, model, INTEGER_MODEL, new Integer(0));
+                return;
+            }
+            if (upperCaseModel.startsWith(LONG_MODEL)) {
+                checkPrimitiveValueInRange(task, variable, model, LONG_MODEL, new Long(0));
+                return;
+            }
+            if (upperCaseModel.startsWith(DOUBLE_MODEL)) {
+                checkPrimitiveValueInRange(task, variable, model, DOUBLE_MODEL, new Double(0));
+                return;
+            }
+            if (upperCaseModel.startsWith(FLOAT_MODEL)) {
+                checkPrimitiveValueInRange(task, variable, model, FLOAT_MODEL, new Float(0));
+                return;
+            }
+            if (upperCaseModel.startsWith(SHORT_MODEL)) {
+                checkPrimitiveValueInRange(task, variable, model, SHORT_MODEL, new Short((short) 0));
+                return;
+            }
+
+            if (upperCaseModel.startsWith(DATETIME_MODEL)) {
+                checkDateTime(task, variable, model);
+                return;
+            }
+
+            if (upperCaseModel.startsWith(LIST_MODEL)) {
+                checkMemberOfList(task, variable, model);
+                return;
+            }
+
+            if (upperCaseModel.startsWith(REGEXP_MODEL)) {
+                checkRegexp(task, variable, model);
+                return;
+            }
+
+            if (upperCaseModel.startsWith(MODEL_FROM_URL_MODEL)) {
+                checkModelFromUrl(task, variable, model);
+                return;
+            }
+        }
+    }
+
+    private void checkBoolean(Task task, TaskVariable variable, String model) throws JobCreationException {
+        String value = getValueAndCheckNull(task, variable, model);
+        boolean isBoolean;
+        switch (value.toLowerCase()) {
+            case "0":
```

Comment:
Allowing too many representations leads to confusion... could simply use `Boolean.valueOf(...)`.